define/securityHelpers.go

```diff
@@ -101,48 +107,48 @@ func VerifySignature(publicKey string, data []byte, signature string) error {
 	if len(publicKey) != ed.PublicKeySize {
 		return ErrRequestDataInvalid
 	}
-	//check the validity of the signature
-	//TODO method is defined supposing args are encrypted
-	//the key is in base64 so we need to decrypt it before using it
-	keyDecoded, err := Decode(publicKey)
-	if err != nil {
-		return ErrEncodingFault
+	if ed.Verify([]byte(publicKey), data, []byte(signature)) {
+		return nil
 	}
-	//data is also in base64 so we need to decrypt it before using it
-	dataDecoded, err := Decode(string(data))
-	if err != nil {
-		return ErrEncodingFault
+	//invalid signature
+	return ErrRequestDataInvalid
+}
+
+/*
+	we check that Sign(sender||data) is the given signature
+*/
+func VerifyWitnessSignature(publicKey string, data []byte, signature string) error {
+	//check the size of the key as it will panic if we plug it in Verify
+	if len(publicKey) != ed.PublicKeySize {
+		return ErrRequestDataInvalid
 	}
-	if ed.Verify(keyDecoded, dataDecoded, []byte(signature)) {
+
+	if ed.Verify([]byte(publicKey), data, []byte(signature)) {
 		return nil
 	}
 	//invalid signature
 	return ErrRequestDataInvalid
 }
 
-//TODO be careful about the size and the order !
-/*Maybe have a fixed size byte ?
-To handle checks while the slice is in construction, the slice must have full space
-from the beginning. We should check how to create fixed length arrays in go. And
-instead of appending in witness_message, put them in the slot which matches the slot
-of the witness id in witness[]
-
-	Witness[1,2,3...]
-	witnessSignature[_,_,_./.]
-	WitnessSignatures[3,6,2,1]
+/*
+	handling of dynamic updates with object as item and not just string
+	*publicKeys is already decoded
+    *sender and signature are not already decoded
 */
-func VerifyWitnessSignatures(publicKeys []byte, signatures []byte, data string, sender string) error {
+func VerifyWitnessSignatures(publicKeys []string, witnessSignaturesEnc []string, sender string) error {
 	senderDecoded, err := Decode(sender)
 	if err != nil {
 		return ErrEncodingFault
 	}
-	dataDecoded, err := Decode(data)
-	if err != nil {
-		return ErrEncodingFault
-	}
-	toCheck := append(senderDecoded, dataDecoded...)
-	for i := 0; i < len(signatures); i++ {
-		err := VerifySignature(string(publicKeys[i]), toCheck, string(signatures[i]))
+	//TODO do we only check the pairs in witnessSignaturesEnc (1) or do need to verify that the publicKey
+	// of the pair is in the publicKeys before (2)?
```

Comment:
TODO unclear for me :)

Reply:
I did clarify it a bit but the TODO concerns mostly the developpers :)